### PR TITLE
(fix): O3-4085 show dialog to user before discarding new appointment changes.

### PR DESCRIPTION
**Pull request by amosmachora**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR adds the ability for the user to be notified if they try exiting the new appointments workspace with unsaved changes.

## Screenshots
https://github.com/user-attachments/assets/c905fa25-86f0-478b-a945-9313d70112b9

## Related Issue
https://openmrs.atlassian.net/issues/O3-4085

### CODE REVIEW

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx [hunk 1]
```diff
@@ -33,17 +28,14 @@ import {
...
-  checkAppointmentConflict,
-  saveAppointment,
-  saveRecurringAppointments,
-  useAppointmentService,
-  useMutateAppointments,
-} from './appointments-form.resource';
-import { useProviders } from '../hooks/useProviders';
-import type { Appointment, AppointmentPayload, RecurringPattern } from '../types';
+import dayjs from 'dayjs';
+import React, { useContext, useEffect, useState } from 'react';
```
Comment (brandones) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx:35]:
```suggestion
```

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx [hunk 2]
```diff
@@ -1,6 +1,3 @@
-import React, { useContext, useEffect, useState } from 'react';
```
Comment (brandones) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx:1]: React at the top is idiomatic.

**Review by brandones — APPROVED** [review]
Good change, thank you!

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx (hunk 2 above)
Comment (amosmachora) [reply packages/esm-appointments-app/src/form/appointments-form.component.tsx:1]: oh, it's prettier that auto-formatted it that way I really couldn't care less where the react import lives.

**Pull request merged**